src/meta/src/manager/catalog/fragment.rs

```diff
@@ -775,6 +776,104 @@ impl FragmentManager {
         actor_maps
     }
 
+    // edit the `rate_limit` of the `Source` node in given `source_id`'s fragments
+    // return the actor_ids to be applied
+    pub async fn update_source_rate_limit_by_source_id(
+        &self,
+        source_id: SourceId,
+        rate_limit: Option<u32>,
+    ) -> MetaResult<HashMap<FragmentId, Vec<ActorId>>> {
+        let map = &mut self.core.write().await.table_fragments;
+        let mut table_id_to_apply = HashSet::new();
+        'table: for (table_id, table_fragments) in map.iter() {
+            for fragment in table_fragments.fragments.values() {
+                for actor in &fragment.actors {
+                    if let Some(stream_node) = actor.nodes.as_ref() {
+                        if let Some(NodeBody::Source(ref node)) = stream_node.node_body.as_ref() {
+                            if let Some(node_inner) = &node.source_inner && node_inner.source_id == source_id {
+                                table_id_to_apply.insert(*table_id);
+                                continue 'table;
```

Comment:
Why don't we directly update source_limit here?
```rs
                        if let Some(NodeBody::Source(ref mut node)) = stream_node.node_body.as_mut()
                        {
                            if let Some(source_inner) = &mut node.source_inner {
                                source_inner.rate_limit = rate_limit;
                            }
                            actor_to_apply.push(actor.actor_id);
                        }
```

Reply:
`BTreeMapTransaction` wrapped table fragments do not impls `Iter`, so I must know the related `table_id`s ahead of time.

Reply:
A bit ugly here. Why not just add an `iter` for it?